src/compas/datastructures/mesh/core/mesh.py

```
@@ -1371,25 +1375,56 @@ def vertices_on_boundaries(self):
         if not vertices_all:
             return vertices_all
 
-        key = sorted([(key, self.vertex_coordinates(key)) for key in vertices_all], key=lambda x: (x[1][1], x[1][0]))[0][0]
+        special = []
+        for key in vertices_all:
+            count = 0
+            for nbr in self.vertex_neighbors(key):
+                face = self.halfedge_face(key, nbr)
+                if face is None:
+                    count += 1
+            if count > 1:
+                special.append(key)
```

If this were included in the `for nbr in self.vertex_neighrbors(key)` loop and ended with a `break`, it might fail faster.  Well, not "fail", but...